### PR TITLE
[Runner]: initial setup for testrunner

### PR DESCRIPTION
**Pull request by 0xE282B0**
Hi,
since this is the first testrunner I'm working on I'd appreciate some feedback 😊

### CODE REVIEW

**Comment by KTSnowy** [comment]
Would it be alright to merge this and test it on the website? I'll open a PR on the other repo to turn on the test-runner

**Comment by iHiD** [comment]
It needs to be tested carefully before it's deployed, as it hits our production infrastructure at that stage, and deployment also has some hoops we have to jump through. @ErikSchierboom will review it before we merge :)

**Comment by ErikSchierboom** [comment]
Great work! The tests are running nicely.
As a second test, I've just run the test runner on the `leap` exercise from the cobol repo using the `proof.ci.cob` file as the input, but that fails without a message. The created `results.json` looks like this:

```json
{
  "version": 1,
  "status": "fail",
  "message": ""
}
```

Maybe the exercises in the `cobol` repo use a different setup/files/configuration than the ones used for testing?

**Comment by 0xE282B0** [comment]
@ErikSchierboom, yes you're right. The `test.sh` scripts are not in sync.
And the `sanitized_test_output` seems to be too aggressive. I'll fix that.

**Comment by KTSnowy** [comment]
Hey @0xE282B0 I merged the PR exercism/cobol#44

**Comment by 0xE282B0** [comment]
@ErikSchierboom, with these changes, the tests are running for me.

**Comment by ErikSchierboom** [comment]
@0xE282B0 Well, yes, the tests within this repo are working, but the test runner on the website does not yet work unfortunately. We do get a different error:

```json
{
  "version": 1,
  "status": "fail",
  "message": "/opt/test-runner/bin/run.sh: 37: ./tests/leap/test.sh: not found  \nSOMETHING WENT WRONG DURING TEST SETUP, PLEASE OPEN A TICKET AT: https://github.com/exercism/cobol/issues/new"
}
```

This is caused by the path to `test.sh` being hardcoded to the test path: https://github.com/exercism/cobol-test-runner/blob/feat/testrunner/bin/run.sh#L37 What the test runner should use is the fact that the second argument (stored in `$solution_dir`) contains the exercise's directory, which in turn contains the `test.sh` directory. 

Changing the above-mentioned line to `test_output=$(${solution_dir}/test.sh 2>&1)` fixes the above issue, but then I get:

```json
{
  "version": 1,
  "status": "fail",
  "message": "cobc: test.cob: No such file or directory"
}
```

The way that I'm testing this is by running the test runner using docker on the `leap` exercise in the `cobol` repo using (for my machine):

```shell
./bin/run-in-docker.sh leap /home/erik/exercism/cobol/exercises/practice/leap /home/erik/exercism/cobol/exercises/practice/leap
```

**Comment by 0xE282B0** [comment]
Thanks @ErikSchierboom,
Good to know, I somehow thought that the exercise would be mounted into the tests folder. But this way it makes a lot more sense. With https://github.com/exercism/cobol/pull/48 the exercises are working for me with the test command you provided.

File: bin/run.sh [hunk 1]
```diff
@@ -34,7 +34,7 @@ echo "${slug}: testing..."
 # Run the tests for the provided implementation file and redirect stdout and
 # stderr to capture it
 # TODO: Replace 'RUN_TESTS_COMMAND' with the command to run the tests
-test_output=$(RUN_TESTS_COMMAND 2>&1)
+test_output=$(${solution_dir}/test.sh ${slug} 2>&1)
```
Comment (ErikSchierboom) [inline bin/run.sh]: I still got an error, which I could fix by doing:

```suggestion
test_output=$(bash ${solution_dir}/test.sh ${slug} 2>&1)
```

Comment (0xE282B0) [reply bin/run.sh]: Right.

**Review by ErikSchierboom — APPROVED** [review]
It now works on my machine! 🥳

**Comment by 0xE282B0** [comment]
Woo Hoo, thanks for your patience. 😊

**Comment by ErikSchierboom** [comment]
We'll need to do one thing for it to be working correctly. I'll create the corresponding PR.

**Comment by ErikSchierboom** [comment]
PR opened: https://github.com/exercism/terraform/pull/77 

@iHiD are the docker/aws secrets available to this repo?

**Comment by iHiD** [comment]
They are now.

**Comment by ErikSchierboom** [comment]
@0xE282B0 This is now ready to be merged! I'll let you do the honors.

**Pull request merged**